src/redis/cache_impl.go

```diff
@@ -18,12 +18,12 @@ import (
 )
 
 type rateLimitCacheImpl struct {
-	pool Pool
-	// Optional Pool for a dedicated cache of per second limits.
-	// If this pool is nil, then the Cache will use the pool for all
-	// limits regardless of unit. If this pool is not nil, then it
+	client Client
+	// Optional Client for a dedicated cache of per second limits.
+	// If this client is nil, then the Cache will use the client for all
```

Comment:
Might be not related to your change. 
But the description seems not right regarding `if this client is nil`. I think we need to specify it as at least a pointer, then we can determine whether it is nil or not.

Reply:
It is an interface, should be OK for nil testing.